netbox/netbox/forms/base.py

```diff
@@ -71,6 +76,15 @@ def _get_custom_fields(self, content_type):
     def _get_form_field(self, customfield):
         return customfield.to_form_field(for_csv_import=True)
 
+    def clean_tags(self):
+        data = self.cleaned_data['tags']
+        existing_tags = Tag.objects.values_list('slug', flat=True)
```

Comment:
This is less than ideal as it will query all tags from the database for each row being imported. Instead of using TagField in the CSV form, we should use a model choice field, which handles the validation of existing objects natively. We have CSVModelChoiceField for individual objects; the same approach can likely be used for selecting multiple objects. I'll take a look.

Reply:
Introduced CSVModelMultipleChoiceField to handle this. Will be suitable for the assignment of other M2M relationships as well.